File: java-recipes/src/main/java/com/fargopolis/models/KnownSpell.java
package com.fargopolis.models;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class KnownSpell {
    private Integer characterId;
    private String spellName;
    private String spellKey;
    private Integer spellLevel;
}

File: recipe-site/app/_models/KnownSpell.ts
export default class KnownSpell {
    spellName: string;
    characterId: number;
    spellKey: string;
    spellLevel: number;

    constructor(spellName: string, spellKey: string, characterId: number, spellLevel: number) {
        this.spellName = spellName;
        this.spellKey = spellKey;
        this.characterId = characterId;
        this.spellLevel = spellLevel;
    }
}